client/landing/stepper/declarative-flow/flows/onboarding/onboarding.ts

```diff
@@ -228,72 +180,6 @@ const onboarding: Flow = {
 
 		const submit = async ( providedDependencies: ProvidedDependencies = {} ) => {
 			switch ( currentStepSlug ) {
-				case 'goals': {
-					const goalsUrl = withLocale( '/setup/onboarding/goals', locale );
-					const { intent } = providedDependencies;
-
-					switch ( intent ) {
-						case SiteIntent.Import: {
-							const migrationFlowLink = withLocale( '/setup/hosted-site-migration', locale );
-							return window.location.assign(
-								addQueryArgs( migrationFlowLink, {
-									back_to: goalsUrl,
-								} )
-							);
-						}
-
-						case SiteIntent.DIFM:
-							return navigate( 'difmStartingPoint' );
-
-						default: {
-							if ( isDesignChoicesStepEnabled ) {
-								return navigate( 'design-choices' );
-							}
-							return navigate( 'design-setup' );
-						}
-					}
-				}
-
-				case 'design-setup': {
-					return navigate( 'domains' );
-				}
-
-				case 'design-choices': {
-					// __a8cBigSkyOnboarding is set as a hack so that the @automattic/calypso-products can know what the users
-					// selection was. Accessing the data store is tricky from there.
-					// See is-big-sky-onboarding.ts
-					if ( providedDependencies.destination === 'launch-big-sky' ) {
-						setCreateWithBigSky( true );
-						return navigate( 'domains' );
-					}
-					setCreateWithBigSky( false );
-					return navigate( providedDependencies.destination as string );
-				}
-
-				case 'difmStartingPoint': {
-					const difmFlowLink = addQueryArgs(
-						withLocale( '/start/website-design-services', locale ),
-						{
-							back_to: window.location.href.replace( window.location.origin, '' ),
-						}
-					);
-
-					if ( isUserLoggedIn ) {
-						return window.location.assign( difmFlowLink );
-					}
-
-					const loginUrl = getLoginUrl( {
-						variationName: flowName,
-						redirectTo: difmFlowLink,
-						locale,
-						extra: {
-							back_to: window.location.href.replace( window.location.origin, '' ),
-						},
-					} );
-
-					return window.location.assign( loginUrl );
-				}
-
```

Comment:
We can remove this code becuase the related steps are only included when `isGoalsAtFrontExperiment` is `true` (which is never the case)